Findingcircles.py

```diff
@@ -0,0 +1,371 @@
+        ax1.imshow(cutoutSmoothed, origin='lower')
+        ax2.plot(cutoutSmoothed[halfbox, :])
+
+    # We normalize and remove background sky values:
+    normValue = np.percentile(cutoutSmoothed, config["normPercent"])
+    skyValue = np.percentile(cutoutSmoothed, config["skyPercent"])
+
+    normImage = (cutoutSmoothed-skyValue)/normValue
+
+    if doPlot:
+        path = os.path.join(path, "detail3.png")
+        ax3.imshow(normImage, origin='lower')
+        ax4.plot(normImage[halfbox, :])
+        fig.savefig(path)
+
+    return normImage, cutoutSmoothed
+
+
+def _detectMask(normImage, config, path, doPlot):
+    # Adding the option that we can either use a predefined value,
+    # or we can use an automatically calculated clip.
+    if "maxclip" in config.keys():
+        clip = config["maxclip"]
+    else:
+        max_img = np.max(normImage)
+        mean_img = np.mean(normImage)
+        clip = max_img/2 - mean_img
+    normMask = ma.getmask(ma.masked_greater_equal(normImage, clip))
+    intImage = np.uint8(255*normMask)
+
+    if doPlot:
+        path = os.path.join(path, "detail4.png")
+        fig = plt.figure()
+        plt.imshow(intImage, origin='lower', cmap='RdGy')
+        plt.colorbar()
+        fig.savefig(path)
+
+    return intImage
+
+
+def _applyHoughTransform(intImage, min_dist, params):
+    circle = cv2.HoughCircles(intImage, cv2.HOUGH_GRADIENT, 1, min_dist, **params)
+
+    circle = np.round(circle[0, :]).astype(int)
+    return circle
+
+
+def _planeskew(smoothedImage, normImage, config, path, doPlot):
+    """ What if we wanted to also find the skew, of the flux plane? Here we
+    attempt to fit a plane to the normalized fluxes, so in case we do see that
+    the flux is uneven across the image, we have the coefficients for a
+    z = a*x + b*y + c plane. with z being the flux.
+    This implementation is heavily inspired by the example given in
+    https://gist.github.com/amroamroamro/1db8d69b4b65e8bc66a6
+    """
+    if "maxclip" in config.keys():
+        clip = config["maxclip"]
+    else:
+        max_img = np.max(normImage)
+        mean_img = np.mean(normImage)
+        clip = max_img/2 - mean_img
+    nmi = ma.masked_less_equal(normImage, clip)
+
+    if doPlot:
+        path = os.path.join(path, "cutout.png")
+        fig = plt.figure()
+        plt.imshow(nmi, origin='lower')
+        plt.colorbar()
+        fig.savefig(path)
+
+    grid = np.indices(nmi.shape)
+    mask = ma.getmask(nmi)
+    y = ma.array(grid[0], mask=mask)
+    x = ma.array(grid[1], mask=mask)
+    z = ma.array(normImage, mask=mask)
+    data = np.c_[x.compressed(), y.compressed(), z.compressed()]
+    a = np.c_[data[:, 0], data[:, 1], np.ones(data.shape[0])]
+    import scipy.linalg as linalg
+    c, _, _, _ = linalg.lstsq(a, data[:, 2])
+    return c
+
+
+def _getEfdData(client, dataSeries, startTime, endTime):
+    import asyncio
+    import nest_asyncio
+    nest_asyncio.apply()
```

Comment:
Definitely need a comment saying why we need this magic line! It's crucial, and functionally magic without a comment.